harvester/src/run_daily_weather.py

```diff
@@ -0,0 +1,158 @@
+        if data_point[field] is not None
+    ]
+
+
+print(f"📅 Fetching weather data for {len(date_list)} days...")
+weather_days_in_db = []
+with database_connection.cursor() as cur:
+    cur.execute("SELECT measure_day, day_finished FROM daily_weather_data;")
+    weather_days_in_db = cur.fetchall()
+
+for date in date_list:
+    today = datetime.date.today()
+    full_day = date
+
+    existing_weather_in_db_for_this_day = [
+        data_point_in_db
+        for data_point_in_db in weather_days_in_db
+        if data_point_in_db[0].date() == full_day
+    ]
+    if existing_weather_in_db_for_this_day != []:
+        logging.info(f"🌦 Weather data for {full_day} already exists in the database...")
+        if existing_weather_in_db_for_this_day[0][1] == False:
+            logging.info(
+                f"🌦 Weather data for {full_day} was not finished in last run, updating now..."
+            )
+            with database_connection.cursor() as cur:
+                cur.execute(
+                    "DELETE FROM daily_weather_data WHERE measure_day = %s", [today]
+                )
+            database_connection.commit()
+        else:
+            continue
+
+    # Using BrightSky API to fetch weather data https://brightsky.dev/docs/#/
+    # Hint: No API key is required
+    url = "https://api.brightsky.dev/weather"
+    params = {
+        "date": full_day,
+        "lat": WEATHER_HARVEST_LAT,
+        "lon": WEATHER_HARVEST_LNG,
+    }
+    headers = {"Accept": "application/json"}
+    response = requests.get(url, params=params, headers=headers)
+    weather_raw = response.json()
+    weather = weather_raw["weather"]
+
+    # Aggregate hourly weather data to daily weather data
+    sum_precipitation_mm_per_sqm = sum(extract(weather, "precipitation"))
+    avg_temperature_celsius = sum(extract(weather, "temperature")) / len(weather)
+    avg_pressure_msl = sum(extract(weather, "pressure_msl")) / len(weather)
+    sum_sunshine_minutes = sum(extract(weather, "sunshine"))
+    avg_wind_direction_deg = sum(extract(weather, "wind_direction")) / len(weather)
+    avg_wind_speed_kmh = sum(extract(weather, "wind_speed")) / len(weather)
+    avg_cloud_cover_percentage = sum(extract(weather, "cloud_cover")) / len(weather)
+    avg_dew_point_celcius = sum(extract(weather, "dew_point")) / len(weather)
+    avg_relative_humidity_percentage = sum(extract(weather, "relative_humidity")) / len(
+        weather
+    )
+    avg_visibility_m = sum(extract(weather, "visibility")) / len(weather)
+    avg_wind_gust_direction_deg = sum(extract(weather, "wind_gust_direction")) / len(
+        weather
+    )
+    avg_wind_gust_speed_kmh = sum(extract(weather, "wind_gust_speed")) / len(weather)
+
+    source_dwd_station_ids = extract(weather_raw["sources"], "dwd_station_id")
+
+    day_finished = full_day < today
+
+    logging.info(f"🌦 Weather data for {full_day} fetched via BrightySky API...")
+
+    with database_connection.cursor() as cur:
```

Comment:
So this does a write for every date in date_list? Could we aggregate this into a `INSERT INTO daily_weather_data (fields...) VALUES(vars...),(...),(...)` to not have to many execute calls? Or does the  `database_connection.commit` that?

Reply:
the commit does it. but it was at the wrong intendation, fixed in last commit.